### PR TITLE
MISUV-6034-update-api-1171-new-accounting-method-response

### PR DESCRIPTION
**Pull request by simondrugan16**

### CODE REVIEW

**Comment by avelx** [comment]
@simondrugan16  - can you address platops-pr-bot warning:
`Remove println("UUUUUUUUUU" + user.incomeSources) from [app/controllers/incomeSources/manage/ManageIncomeSourceDetailsController.scala](https://github.com/hmrc/income-tax-view-change-frontend/blob/MISUV-6034-update-api-1171-new-accounting-method-response/app/controllers/incomeSources/manage/ManageIncomeSourceDetailsController.scala#L116)
Remove println("OOOOOOOOO" + viewModel) from`

File: app/controllers/incomeSources/add/BusinessAccountingMethodController.scala [hunk 1]
```diff
@@ -56,24 +56,28 @@ class BusinessAccountingMethodController @Inject()(val authenticate: Authenticat
                                               (implicit user: MtdItUser[_], backUrl: String, postAction: Call, messages: Messages): Future[Result] = {
     val userActiveBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses.filterNot(_.isCeased)
 
-    if (userActiveBusinesses.flatMap(_.cashOrAccruals).distinct.size > 1) {
+    if (userActiveBusinesses.flatMap(_.cashOrAccrualsFlag).distinct.size > 1) {
```
Comment (avelx) [inline app/controllers/incomeSources/add/BusinessAccountingMethodController.scala:59]: I am not sure why we use distinct over userActiveBusinesse, pls remove this, as we expecting data to be distinct here anyway

Comment (avelx) [reply app/controllers/incomeSources/add/BusinessAccountingMethodController.scala:59]: cool, make sense, thanks

File: app/controllers/incomeSources/add/BusinessAccountingMethodController.scala [hunk 2]
```diff
@@ -56,24 +56,28 @@ class BusinessAccountingMethodController @Inject()(val authenticate: Authenticat
...
       Logger("application").error(s"${if (isAgent) "[Agent]"}" +
-        s"Error getting business cashOrAccrualsField")
+        s"Error getting business cashOrAccrualsFlag Field")
     }
 
     userActiveBusinesses match {
-      case head :: _ if (head.cashOrAccruals.isDefined) =>
-        val accountingMethod: String = head.cashOrAccruals.get
+      case head :: _ if (head.cashOrAccrualsFlag.isDefined) =>
+        val accountingMethod: String = if (head.cashOrAccrualsFlag.get) {
```
Comment (avelx) [inline app/controllers/incomeSources/add/BusinessAccountingMethodController.scala:66]: if you can move this type transformation to the level of relevant case class => BusinessDetailsModel
boolean => String

File: app/views/incomeSources/manage/ManageIncomeSourceDetails.scala.html [hunk 3]
```diff
@@ -172,8 +172,8 @@
             <dd class="govuk-summary-list__value" id="business-accounting-method">@{
                 viewModel.businessAccountingMethod match {
                     case Some(x) => x match {
-                        case value if value == "cash" => getMessage("cash-accounting")
-                        case value if value == "accruals" => getMessage("traditional-accounting")
+                        case value if !value => getMessage("cash-accounting")
```
Comment (avelx) [inline app/views/incomeSources/manage/ManageIncomeSourceDetails.scala.html:175]: you can move this code to the level of viewModel, ie. create method which will return key
so you call call like:
viewModel.businessAccountingMethodAsKey(  getMessage(_) )

**Review by Joshtest1570 — APPROVED** [review]

**Pull request merged**